tap_google_ads/client.py

```diff
@@ -1,17 +1,104 @@
+import os
+import json
+import functools
+from google.auth.transport.requests import Request
+from requests import Session
 from google.ads.googleads.client import GoogleAdsClient
+from google.oauth2 import service_account
 
+def _initialize_credentials_decorator(func):
+    """A decorator used to easily initialize credentials objects.
+    Returns:
+        An initialized credentials instance
+    """
 
-def create_sdk_client(config, login_customer_id=None):
-    CONFIG = {
-        "use_proto_plus": False,
-        "developer_token": config["developer_token"],
-        "client_id": config["oauth_client_id"],
-        "client_secret": config["oauth_client_secret"],
-        "refresh_token": config["refresh_token"],
-    }
+    @functools.wraps(func)
+    def initialize_credentials_wrapper(*args, **kwargs):
+        credentials = func(*args, **kwargs)
+        # If the configs contain an http_proxy, refresh credentials through the
+        # proxy URI
+        proxy = kwargs.get("http_proxy")
+        if proxy:
+            session = Session()
+            session.proxies.update({"http": proxy, "https": proxy})
+            credentials.refresh(Request(session=session))
+        else:
+            credentials.refresh(Request())
+        return credentials
+
+    return initialize_credentials_wrapper
```

Comment:
Can we try to import this function from the https://github.com/googleads/google-ads-python/blob/0f0fde31d39c0b99c91112b86462d1ffd2e912cf/google/ads/googleads/oauth2.py#L30

```python
from google.ads.googleads.oauth2 import _initialize_credentials_decorator
```